src/vs/workbench/contrib/terminal/browser/terminalTypeAheadAddon.ts

```diff
@@ -1291,6 +1291,12 @@ export class TypeAheadAddon extends Disposable implements ITerminalAddon {
 		}));
 	}
 
+	// We need a new reference to the Terminal object because the buffer appears to be
+	// corrupted.
+	public reset(terminal: Terminal) {
+		this.timeline = new PredictionTimeline(terminal, this.typeaheadStyle!);
```

Comment:
This is the workaround. If not for this, the terminal.buffer used in `onUserData(..)` (which comes from `this.timeline.terminal.buffer`) has values of 0 for all Y props.